components/proxy/src/main/java/com/hotels/styx/admin/handlers/CurrentRequestsHandler.java

```diff
@@ -0,0 +1,163 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.admin.handlers;
+
+import static com.google.common.net.MediaType.PLAIN_TEXT_UTF_8;
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_TYPE;
+import static com.hotels.styx.api.HttpResponseStatus.OK;
+import static java.lang.management.ManagementFactory.getThreadMXBean;
+
+import java.lang.management.LockInfo;
+import java.lang.management.MonitorInfo;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.nio.charset.Charset;
+
+import com.hotels.styx.api.HttpResponse;
+import com.hotels.styx.api.LiveHttpRequest;
+import com.hotels.styx.api.LiveHttpResponse;
+import com.hotels.styx.common.http.handler.BaseHttpHandler;
+import com.hotels.styx.server.track.CurrentRequestTracker;
+
+/**
+ * Admin handler that will help in tracking only the current HTTP requests to Styx.
+ */
+public class CurrentRequestsHandler extends BaseHttpHandler {
+
+    private static final Charset UTF_8 = Charset.forName("UTF-8");
+    private final ThreadMXBean threadMXBean;
+
+    public CurrentRequestsHandler() {
+        this.threadMXBean = getThreadMXBean();
+    }
+
+    @Override
+    public LiveHttpResponse doHandle(LiveHttpRequest request) {
+        boolean withStackTrace = request.queryParam("withStackTrace").isPresent() && "true".equalsIgnoreCase(request.queryParam("withStackTrace").get());
+        return HttpResponse
+                .response(OK)
+                .disableCaching()
+                .header(CONTENT_TYPE, PLAIN_TEXT_UTF_8)
+                .body(getCurrentRequestContent(withStackTrace), UTF_8, true)
+                .build()
+                .stream();
+    }
+
+    private String getCurrentRequestContent(boolean withStackTrace) {
```

Comment:
I would tweak the output bit more:

For proxied requests, which are waiting for origin responses, it currently says:

```
   ...   Thread Info: ...
Thread stack not available: Waiting response from origin.]
```

This looks bit funny as the response state is embedded in the "Thread Info". But in fact the "thread info" actually depends on the request state: The stack trace is relevant only when the response is going through the plugin pipeline. Therefore, let's promote a request state to be displayed *before* the stack trace.

I would remove the stack trace altogether when the request is waiting for a response from the origins. Just display this:

```
Request state: Waiting for response from origin.
```

When a stack trace is relevant (requests in a proxy pipeline), I would include it *after* the request state:

```
Request state: plugins pipeline.
Thread Info:
Thread stack not available: Waiting response from origin.]
```

In fact, even better if this could be formatted out structurally as a JSON string.